roles/undercloud-install/templates/introspect.sh.j2

```
@@ -107,6 +107,18 @@ clean()
                                 on_count=$(ironic --json node-list|jq -r ".[]| select(.[\"provision_state\"] == \"clean wait\")| .[\"uuid\"]" | wc -l) ; \
                             done'
     set -e
+
+    for node in `ironic --json node-list | jq -r '.[]| select(.["provision_state"] == "clean failed")|.["uuid"]'`; do
+        ironic node-set-provision-state $node manage
+        sleep 10
+    done
+    for node in `ironic --json node-list | jq -r '.[]| select(.["provision_state"] == "clean wait")|.["uuid"]'`; do
+        ironic node-set-maintenance $node true
```

Can we use openstack baremetal api please?

---

This isn't part of my change, it is a squash error